src/System.Linq/tests/SkipWhileTests.cs

```diff
@@ -0,0 +1,181 @@
+                    select x;
+
+            Assert.Equal(q.SkipWhile(x => true), q.SkipWhile(x => true));
+        }
+
+        [Fact]
+        public void SameResultsRepeatCallsStringQuery()
+        {
+            var q = from x in new[] { "!@#$%^", "C", "AAA", "", "Calling Twice", "SoS", String.Empty }
+                    where !String.IsNullOrEmpty(x)
+                    select x;
+
+            Assert.Equal(q.SkipWhile(x => true), q.SkipWhile(x => true));
+        }
+
+        [Fact]
+        public void PredicateManyFalseOnSecond()
+        {
+            int[] source = { 8, 3, 12, 4, 6, 10 };
+            int[] expected = { 3, 12, 4, 6, 10 };
+
+            Assert.Equal(expected, source.SkipWhile((e, i) => e % 2 == 0));
+        }
+
+        [Fact]
+        public void PredicateManyFalseOnSecondInex()
+        {
+            int[] source = { 8, 3, 12, 4, 6, 10 };
+            int[] expected = { 3, 12, 4, 6, 10 };
+            
+            Assert.Equal(expected, source.SkipWhile((e, i) => e % 2 == 0));
+        }
+
+        [Fact]
+        public void PredicateTrueOnSecondFalseOnFirstAndOthers()
+        {
+            int[] source = { 3, 2, 4, 12, 6 };
+            int[] expected = { 3, 2, 4, 12, 6 };
+
+            Assert.Equal(expected, source.SkipWhile((e, i) => e % 2 == 0));
+        }
+
+        [Fact]
+        public void PredicateTrueOnSecondFalseOnFirstAndOthersIndex()
+        {
+            int[] source = { 3, 2, 4, 12, 6 };
+            int[] expected = { 3, 2, 4, 12, 6 };
+
+            Assert.Equal(expected, source.SkipWhile((e, i) => e % 2 == 0));
+        }
+
+        [Fact]
+        public void FirstExcludedByIndex()
+        {
+            int[] source = { 6, 2, 5, 3, 8 };
+            int[] expected = { 2, 5, 3, 8 };
+
+            Assert.Equal(expected, source.SkipWhile((element, index) => index == 0));
+        }
+
+        [Fact]
+        public void AllButLastExcludedByIndex()
+        {
+            int[] source = { 6, 2, 5, 3, 8 };
+            int[] expected = { 8 };
+
+            var actual = source.SkipWhile((element, index) => index < source.Length - 1);
+        }
+        
+        private sealed class FastInfiniteEnumerator : IEnumerable<int>, IEnumerator<int>
+        {
+            public IEnumerator<int> GetEnumerator()
+            {
+                return this;
+            }
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return this;
+            }
+            public bool MoveNext()
+            {
+                return true;
+            }
+            public void Reset()
+            {
+            }
+            object IEnumerator.Current
+            {
+                get { return 0; }
+            }
+            public void Dispose()
+            {
+            }
+            public int Current
+            {
+                get { return 0; }
+            }
+        }
+
+        [Fact(Skip="Long-running")]
```

Comment:
Yes, [OuterLoop] is used for tests that take a long time or are otherwise difficult to run (e.g. require admin rights).